File: src/lib/cli-install.ts
import { invoke } from "@tauri-apps/api/core";

const CLI_PROMPT_DISMISSED_KEY = "port-watch-cli-prompt-dismissed";

export interface CliInstallStatus {
  installed: boolean;
  linkPath: string;
  targetPath: string | null;
  pointsToApp: boolean;
}

export function isCliInstallPromptDismissed(): boolean {
  try {
    return localStorage.getItem(CLI_PROMPT_DISMISSED_KEY) === "1";
  } catch {
    return false;
  }
}

export function dismissCliInstallPrompt(): void {
  try {
    localStorage.setItem(CLI_PROMPT_DISMISSED_KEY, "1");
  } catch {
    // ignore
  }
}

export async function fetchCliInstallStatus(): Promise<CliInstallStatus | null> {
  try {
    return await invoke<CliInstallStatus>("get_cli_install_status");
  } catch {
    return null;
  }
}

export async function installCliToPath(): Promise<void> {
  await invoke("install_cli_to_path");
}

export async function uninstallCliFromPath(): Promise<void> {
  await invoke("uninstall_cli_from_path");
}

File: src-tauri/src/commands/cli_install.rs
use crate::cli_install::{self, CliInstallStatus};

#[tauri::command]
pub fn get_cli_install_status() -> Result<CliInstallStatus, String> {
    cli_install::get_cli_install_status()
}

#[tauri::command]
pub fn install_cli_to_path() -> Result<(), String> {
    cli_install::install_cli_to_path()
}

#[tauri::command]
pub fn uninstall_cli_from_path() -> Result<(), String> {
    cli_install::uninstall_cli_from_path()
}
